### ISSUE TITLE
Conditional chaining in resulting js

### ISSUE THREAD

**Issue by maasencioh** [cheminfo/analysis#12]
On the building process of storybook I'm having this error

```
ERROR in ./node_modules/base-analysis/lib-esm/util/autoPeakPicking.js 12:45
Module parse failed: Unexpected token (12:45)
You may need an appropriate loader to handle this file type, currently no loaders are configured to process this file. See https://webpack.js.org/concepts#loaders
| export function autoPeakPicking(measurement, options = {}) {
|     const { xVariable = 'x', yVariable = 'y', normalizationOptions, minPeakWidth, gsdOptions = {}, } = options;
>     let x = measurement.variables[xVariable]?.data;
|     let y = measurement.variables[yVariable]?.data;
|     if (!x || !y)
```

I understand that the conditional chaining is not supported, should we build this package in a lower level or is it better to support this from the storybook webpack? @targos @lpatiny

**Comment by targos** [comment]
The optional chaining operator has now been supported by browsers and node for two years, so I don't think we should limit its usage in our libraries.

**Comment by targos** [comment]
That said, the default babel config from storybook is supposed to transpile it already: https://github.com/storybookjs/storybook/blob/next/lib/core-common/src/utils/babel.ts

**Comment by maasencioh** [comment]
Also apparently is [not really supported by this webpack version](https://github.com/webpack/webpack/issues/10227)
> UPD3 fix for optional chaining landed into acorn 7.3.0 but webpack@4 uses acorn 6.x. so fix of this issue will be in webpack@5: #10227 (comment)

I'm trying to see how to add support for this on my local storybook, but I think that is the `"target": "es2019"` what we could change here

**Comment by maasencioh** [comment]
Fixed in repo
